packages/task/src/browser/task-service.ts

```diff
@@ -402,10 +432,146 @@ export class TaskService implements TaskConfigurationClient {
+            return undefined;
+        }
+    }
+
+    async runTasksGraph(task: TaskConfiguration, tasks: TaskConfiguration[], option?: RunTaskOption): Promise<TaskInfo | undefined> {
+        if (task && task.dependsOn && task.dependsOn.length > 0 && task.dependsOrder && task.dependsOrder === 'sequence') {
+            for (let i = 0; i < task.dependsOn.length; i++) {
+                const dependentTask = this.getDependentTask(task.dependsOn[i], tasks);
+                const taskCustomization = await this.getTaskCustomization(dependentTask);
+                const resolvedMatchers = await this.resolveProblemMatchers(dependentTask, taskCustomization);
+                await this.runTasksGraph(dependentTask, tasks, {
+                    customization: { ...taskCustomization, ...{ problemMatcher: resolvedMatchers } }
+                });
+            }
+        } else if (task.dependsOn && task.dependsOn.length > 0 && ((!task.dependsOrder) || (task.dependsOrder && task.dependsOrder === 'parallel'))) {
+            const dependentTasks: { 'task': TaskConfiguration; 'taskCustomization': TaskCustomization; 'resolvedMatchers': ProblemMatcher[] | undefined }[] = [];
+            for (let i = 0; i < task.dependsOn.length; i++) {
+                const dependentTask = this.getDependentTask(task.dependsOn[i], tasks);
+                const taskCustomization = await this.getTaskCustomization(dependentTask);
+                const resolvedMatchers = await this.resolveProblemMatchers(dependentTask, taskCustomization);
+                dependentTasks.push({ 'task': dependentTask, 'taskCustomization': taskCustomization, 'resolvedMatchers': resolvedMatchers });
+            }
+            const promises = dependentTasks.map(item =>
+                this.runTasksGraph(item.task, tasks, {
+                    customization: { ...item.taskCustomization, ...{ problemMatcher: item.resolvedMatchers } }
+                })
+            );
+            await Promise.all(promises);
+        }
+        const taskInfo = await this.runTask(task, option);
+        if (taskInfo) {
+            if (task.isBackground) {
+                return Promise.race([this.getBackgroundTaskEndMatch(taskInfo.taskId), this.getExitCode(taskInfo.taskId)]).then(async result => {
+                    if (typeof result === 'number' && result !== 0) {
+                        await this.throwTaskError('The task: ' + task.label + ' terminated with exit code ' + result + '.');
+                    }
+                    return taskInfo;
+                });
+            } else {
+                const exitCode = await this.getExitCode(taskInfo.taskId);
+                if (exitCode !== 0) {
+                    await this.throwTaskError('The task: ' + task.label + ' terminated with exit code ' + exitCode + '.');
+                }
+            }
+        }
+        return taskInfo;
+    }
+
+    async throwTaskError(errorMessage: string): Promise<void> {
```

Comment:
Is the following method really necessary?

Reply:
fixed, done.